R/pkg/R/types.R

```diff
@@ -94,20 +94,7 @@ checkSchemaInArrow <- function(schema) {
 
   # Both cases below produce a corrupt value for unknown reason. It needs to be investigated.
   field_strings <- sapply(schema$fields(), function(x) x$dataType.toString())
-  if (any(field_strings == "FloatType")) {
```

Comment:
Sounds good. Can we add a test case? I think you can update the test cases with post-fix "- type specification" at test_sparkSQL_arrow.R